### PR TITLE
Eslint airbnb transloadit

### PR DESCRIPTION
**Pull request by mifi**
Use [eslint-config-transloadit](https://github.com/transloadit/eslint-config-transloadit) and fix issues uncovered.
See #90 and https://github.com/transloadit/eslint-config-transloadit/issues/1

### CODE REVIEW

File: test/integration/__tests__/live-api.js [hunk 1]
```diff
@@ -293,9 +294,13 @@ describe('API integration', function () {
       const client = createClient()
 
       let progressCalled = false
-      function onUploadProgress ({ uploadedBytes }) {
+      function onUploadProgress ({ uploadedBytes, totalBytes }) {
         // console.log(uploadedBytes)
         expect(uploadedBytes).toBeDefined()
+        if (isResumable) {
```
Comment (kvz) [inline test/integration/__tests__/live-api.js:300]: these seem unrelated :thinking:

File: test/unit/__tests__/test-pagination-stream.js [hunk 2]
```diff
@@ -1,26 +1,27 @@
-const PaginationStream = require('../../../src/PaginationStream')
 const { Writable } = require('stream')
 const _ = require('lodash')
 
-const toArray = callback => {
+const PaginationStream = require('../../../src/PaginationStream')
```
Comment (kvz) [inline test/unit/__tests__/test-pagination-stream.js:4]: i like the sorting of requires

**Review by kvz — APPROVED** [review]
Few nitpicks/questions, it does look good overall!

One thing, there are quite some cases where you disable eslint rules the code. 

I wonder, if we apply this to the API, if that will be crippling. Like. so many bailouts to add.

But i guess, we can bail out on a per-project level in the .eslintrc, until we move to a reality where all issues can be fixed / explicitly ignored

File: test/integration/__tests__/live-api.js (hunk 1 above)
Comment (mifi) [reply test/integration/__tests__/live-api.js:300]: Yes. I discovered some issues/improvements while converting to the new eslint config. I can pull those commits out to another branch

**Comment by mifi** [comment]
I can comment on the `eslint-disable`s:
- `no-await-in-loop`: This rule is meant to prevent people from looping over a list of things  in series, when it could be run much faster in parallell, with `p-map` etc. Where I disabled this is because I know I want it to run in series. See https://eslint.org/docs/rules/no-await-in-loop
- `no-loop-func`: making functions inside a loop is error prone, because if the user references variables in the loop from the function, they will not have the value that you expect. I agree this one is a bit strict and maybe we should disable it. See https://eslint.org/docs/rules/no-loop-func
- `no-param-reassign`: reassigning function arguments can lead to confusing behavior. I disabled here because the `decorateError` is intentionally not pure as it is meant to mutate the input. https://eslint.org/docs/rules/no-param-reassign
- `no-bitwise` was disabled because in this case we are actually doing a bitwise operation. However in normal JS programming using a bitwise operator is usually a mistake (a typo) which could cause unexpected errors.
- `no-constant-condition` prevents code like `if (true)`, but here it also kicks on `while`.
- `class-methods-use-this` says that the function does not use anything from the class and should then be static or pulled out. but I was lazy to rewrite the unit tests so I wanted to keep it there (mockability)
- `no-console` is good to catch accidental console logs (a library shouldn't do this). But for examples and cli apps etc this can be disabled
- `no-underscore-dangle` is just a style from airbnb eslint that underscores are not to be used (but I disabled for instance method names).

I'm totally fine with completely disabling some or all of these in the central config also

**Comment by mifi** [comment]
I think if we introduce this style guide to the existing code, we can definitely selectively disable certain rules inside .eslintrc and keep that disable list as a todo for each project

File: test/integration/__tests__/live-api.js (hunk 1 above)
Comment (kvz) [reply test/integration/__tests__/live-api.js:300]: No need as long as you are aware :ok_hand:

**Comment by kvz** [comment]
Thanks for the detailed explanation. I think we should go with strict for now, loosen up in individual projects in their own .eslintrc, and based on feedback loosen up in a next version of eslint-transloadit.

Regarding all the extra safety it provides, i'm ok to have many changes. As for stylistic changes, not so much - but that also doesn't seem to be the case looking at this PR :ok_hand:

**Comment by kvz** [comment]
Feel free to merge this one already if you like :tada:

**Comment by kvz** [comment]
or ... do we first want to finalize work on our linting, and then merge all of the open prs .. :thinking:  there may be changes based on findings in the other repos.

**Comment by mifi** [comment]
I think fixing the other repos is a much bigger task, so I think we can merge this one now. If we need to update the config in the future we can create a new PR here for the new version.
